### PR TITLE
add new tests for command key-period-info

### PR DESCRIPTION
**Pull request by saratomaz**
information on the related issue

### CODE REVIEW

File: cardano_node_tests/tests/test_kes.py [hunk 1]
```diff
@@ -132,6 +133,63 @@ def test_expired_kes(
 
     @allure.link(helpers.get_vcs_link())
     @pytest.mark.order(6)
+    @pytest.mark.skipif(
+        not (VERSIONS.cluster_era == VERSIONS.transaction_era == VERSIONS.LAST_KNOWN_ERA),
+        reason="meant to run only with the latest cluster era and the latest transaction era",
+    )
```
Comment (mkoura) [inline cardano_node_tests/tests/test_kes.py:139]: This test should be merged with `test_expired_kes`. This test needs to start cluster instance with specific configuration (same as `test_expired_kes`) and setup and teardown takes ~ 40 mins - it is very time consuming.
Until the merging is done, please disable the test in CI with
```python
    @pytest.mark.skipif(
        bool(os.environ.get("CI")),
        reason="meant to run only outside of CI",
    )
```

File: cardano_node_tests/tests/test_kes.py [hunk 2]
```diff
@@ -175,13 +233,13 @@ def _wait_epoch_chores(this_epoch: int):
             )
 
         with cluster_manager.restart_on_failure():
-            # generate new operational certificate with `--kes-period` in the past
+            # generate new operational certificate with `--kes-period` in the future
             invalid_opcert_file = cluster.gen_node_operational_cert(
-                node_name=node_name,
+                node_name="invalid_opcert_file",
```
Comment (mkoura) [inline cardano_node_tests/tests/test_kes.py:238]: Include `node_name` in the resulting certificate file name.

File: cardano_node_tests/tests/test_kes.py [hunk 3]
```diff
@@ -213,17 +271,22 @@ def _wait_epoch_chores(this_epoch: int):
...
+            kes_period_info = cluster.get_kes_period_info(invalid_opcert_file)
+            check_kes_period_info_result(
+                kes_output=kes_period_info, expected_scenario="invalid_kes_period"
+            )
+
             # generate new operational certificate with valid `--kes-period`
-            opcert_file.unlink()
             valid_opcert_file = cluster.gen_node_operational_cert(
-                node_name=node_name,
+                node_name="valid_opcert_file",
```
Comment (mkoura) [inline cardano_node_tests/tests/test_kes.py:282]: Include `node_name` in the resulting certificate file name.

File: cardano_node_tests/tests/test_kes.py [hunk 4]
```diff
@@ -270,11 +343,12 @@ def test_update_valid_opcert(
         stake_pool_id_dec = helpers.decode_bech32(stake_pool_id)
 
         opcert_file = pool_rec["pool_operational_cert"]
+        opcert_file_old = shutil.copy(opcert_file, f"{opcert_file}_old")
 
         with cluster_manager.restart_on_failure():
             # generate new operational certificate with valid `--kes-period`
             new_opcert_file = cluster.gen_node_operational_cert(
-                node_name=node_name,
+                node_name="new_opcert_file",
```
Comment (mkoura) [inline cardano_node_tests/tests/test_kes.py:351]: Include `node_name` in the resulting certificate file name.

File: cardano_node_tests/tests/kes.py [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+        start_kes_period = metrics.get("qKesStartKesInterval")
+        end_kes_period = metrics.get("qKesEndKesInterval")
+        current_kes_period = metrics.get("qKesCurrentKesPeriod")
+
+        # check if the metrics match with the expected scenario
+        if (
+            node_state_counter <= on_disk_counter
+            and start_kes_period <= current_kes_period <= end_kes_period
+        ):
+            output_scenario = "all_valid"
```
Comment (mkoura) [inline cardano_node_tests/tests/kes.py:26]: Please use constants for the scenarios names both here and in tests. You can create "enum" like
```python
class KesScenarios:
    ALL_VALID = "all_valid"
    ...

# use it like
output_scenario = KesScenarios.ALL_VALID
```

**Review by mkoura — CHANGES_REQUESTED** [review]

File: cardano_node_tests/tests/test_kes.py (hunk 1 above)
Comment (saratomaz) [reply cardano_node_tests/tests/test_kes.py:139]: Totally agree, I tried to merge but it's not straight forward, so I just created a new one to not mess with the running one.

File: cardano_node_tests/tests/test_staking.py [hunk 6]
```diff
@@ -13,6 +14,7 @@
 
 from cardano_node_tests.tests import common
 from cardano_node_tests.tests import delegation
+from cardano_node_tests.tests.kes import check_kes_period_info_result
```
Comment (mkoura) [inline cardano_node_tests/tests/test_staking.py:17]: Change this to module import as well.

File: cardano_node_tests/tests/kes.py [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+            node_state_counter,
+            on_disk_counter,
+            start_kes_period,
+            end_kes_period,
+            current_kes_period,
+        ]
+
+        # check if the metrics match with the expected scenario
+        if (
+            all(type(value) == int for value in metrics_to_compare)
```
Comment (mkoura) [inline cardano_node_tests/tests/kes.py]: Linter errored on this one, more idiomatic is `all(isinstance(value, int)....`

File: cardano_node_tests/tests/kes.py [hunk 8]
```diff
@@ -0,0 +1,56 @@
...
+
+        # check if the metrics match with the expected scenario
+        if (
+            all(isinstance(value, int) for value in metrics_to_compare)
+            and node_state_counter <= on_disk_counter
+            and start_kes_period <= current_kes_period <= end_kes_period
+        ):
+            output_scenario = KesScenarios.ALL_VALID
+    # expected scenario: all_invalid
+    elif not kes_output.get("valid_counters") and not kes_output.get("valid_kes_period"):
```
Comment (mkoura) [inline cardano_node_tests/tests/kes.py:45]: Optional:
`not something and not something_else`
is customary abbreviated as
`not (something or something_else)`

**Review by mkoura — CHANGES_REQUESTED** [review]

**Review by mkoura — APPROVED** [review]
:+1:

**Pull request merged**